main/app.py

```diff
@@ -0,0 +1,352 @@
+            "SELECT COUNT(*) AS office_occupants FROM Persons  WHERE Persons.office_accommodation = ?", [off_name]).fetchall()
+        self.office_occupied = office_space[0][0]
+        return self.office_occupied
+
+    def living_space_count(self, liv_name):
+        '''Keep track of living space allocated should be less than 4'''
+        # print(liv_name)
+        living_space = self.connect.execute(
+            "SELECT COUNT(*) AS living_occupants FROM Persons WHERE Persons.living_accomodation = ?", [liv_name]).fetchall()
+        for liv_space in living_space:
+            self.living_occupied = liv_space[0]
+            print(self.living_occupied)
+        return self.living_occupied
+
+    def add_person(self, first_name, last_name, person_type, want_housing):
+        '''Method receives personnel data and calls other method to save and allocate rooms to personnel '''
+        name = first_name + " " + last_name
+        person_type = person_type
+        want_accommodation = want_housing
+        if person_type.upper() == 'STAFF':
+            self.people_data['Staff'].append(name)
+        else:
+            self.people_data['Fellow'].append({name: want_accommodation})
+
+        self.insert_db(name=name, person_type=person_type,
+                       want_accommodation=want_accommodation)
+
+        self.allocation_rule(name, person_type,
+                             want_accommodation)
+        return name +" " + person_type + " " + want_accommodation
+
+    def get_rooms(self, r_type):
+        return self.connect.execute(
+            "SELECT Name FROM Rooms where type = '" + r_type + "'").fetchall()
+
+    def allocation_rule(self, name, person_type, want_accommodation):
+        '''Randomly allocate everyone an office'''
+        office_rooms = self.get_rooms('O')
+        for office_room in office_rooms:
+            room = Office(self.connect)
+            office_name = office_room[0]
+
+            if not room.is_room_filled(office_name):
+                self.available_office.append(office_name)
+
+        self.office = random.choice(self.available_office)
+        if self.office:
+            self.allocate_office(name, self.office)
+        else:
+            print('No office available')
+
+        if person_type.upper() == 'FELLOW' and want_accommodation == 'Y':
+            living_rooms = self.get_rooms('L')
+            for living_room in living_rooms:
+                room = LivingSpace(self.connect)
+                living_name = living_room[0]
+                if not room.is_room_filled(living_name):
+                    self.available_housing.append(living_name)
+
+            self.housing = random.choice(self.available_housing)
+
+            if self.housing:
+                self.allocate_housing(name, self.housing)
+            else:
+                print("No living space available")
+        else:
+            print("Personnel hasn't requested to be housed")
+
+
+    def insert_db(self, **kwargs):
```

Comment:
**Code Quality: Nitpick**

I feel like you will be able to do a proper implementation of `save_state` if this function were omitted.